packages/runtime-node/src/types.ts

```diff
@@ -134,3 +134,56 @@ export interface IConfigDefinition {
     envName?: string;
     filePath: string;
 }
+
+export interface MetadataCollectionAPI {
+    getRuntimeArguments: () => IEngineRuntimeArguments;
+}
+
+export interface IEngineRuntimeArguments {
+    featureName: string;
+    basePath: string;
+    outputPath: string;
+    configName?: string;
+    devport?: number;
+    nodeEntryPath: string;
+    features: [featureName: string, featureDefinition: Required<IStaticFeatureDefinition>][];
+    config: TopLevelConfig;
+    requiredModules?: string[];
+    runtimeOptions?: StartEnvironmentOptions['options'];
+}
+
+export interface IStaticFeatureDefinition {
+    contextFilePaths?: Record<string, string>;
+    envFilePaths?: Record<string, string>;
+    preloadFilePaths?: Record<string, string>;
+    dependencies?: string[];
+    scopedName: string;
+    resolvedContexts?: Record<string, string>;
+    packageName: string;
+    filePath: string;
+    exportedEnvs?: IEnvironmentDescriptor[];
+}
+
+export const metadataApiToken = {
+    id: 'metadata-api-token',
+};
+
+export const isWorkerThreadEnvStartupMessage = (value: unknown): value is IWorkerThreadEnvStartupMessage => {
```

Comment:
I can't find a place where this type was moved from.. is it a new type?

Reply:
this is new one. not directly related to this particular refactoring, but does not break anything